### DISCUSSION TITLE
Overlay one image onto another and position top left.

### DISCUSSION THREAD

**Discussion by webbooo** [ImageMagick/ImageMagick#5075] [Help] [answered]
Hello,

I have a small project I'm hoping someone could help with.

I have 2 folders and I'm trying to overlay the png's from one folder onto the png's in another with the same number

As an example, folder 'Source' has a range of png files all 75x75 and are numbered from 1 through to 10.

I have a secondary folder 'Destination' with another 10 png's also numbered 1 through to 10.

I'd like the file from C:\source\1.png to overlaid onto the file c:\destination\1.png and be placed in the top left hand corner.

Example pics for demonstration 

file 1.png (C:\source\)

![sample](https://user-images.githubusercontent.com/103241119/164969799-4844ce52-57b5-4d3b-a0de-393c34a26a23.png)

file 1.png (C:\destination\)

![samplea](https://user-images.githubusercontent.com/103241119/164969806-afe14a9c-8ea7-441f-a6dc-382aaab36507.png)

desired outcome

![endresult](https://user-images.githubusercontent.com/103241119/164969817-819a758d-9025-477e-b5d2-d54a4ac63746.png)

Appreciate any help anyone can offer.

Thanks

**Comment by snibgo** [comment]
What version of IM, on what platform? I assume Windows BAT.

First, figure out the IM command that does what you want. Perhaps:
```
magick c:\destination\1.png c:\source\1.png -composite c:\outcome\1.png
```
Then put it in a shell loop. For Windows, I suggest a "for" loop. Type "help for" at the command line for help.

**Reply by webbooo** [reply]
Hi @snibgo 

am using 7.1 on Windows. This command worked thank you, for a single file.   I had a look into loop but most references are for Linux, I didnt quite understand what 'help for' presented but will see what else I can find.

**Reply by webbooo** [reply]
hey @snibgo 

I didnt have any luck trying to loop this command to work through the full folder!  wondering if you had a suggestion or code you could share?  thanks

**Comment by snibgo** [comment, marked as answer]
Something like this (Windows BAT syntax):
```
for %%F in (c:\destination\*.png) do magick %%F c:\source\%%~nF.png -gravity NorthWest -compose Over -composite c:\outcome\%%~nF.png
```
The directory "c:\outcome\" must already exist. Instead of that directory, you can use one of the input directories, but that will overwrite files.

**Reply by webbooo** [reply]
Thanks for this @snibgo  I was able to get this to work for this scenario and on the other thread you replied to, cheers!

Just curious, what can I use to have more control over where the source.png file is placed? Gravity seems to have a few limitations? Can I be specific in terms of how far down and across I'd like the pic to be placed or even rotated?

example

![Capture](https://user-images.githubusercontent.com/103241119/167077233-be062f01-39c7-49c7-8929-717672de4ec7.PNG)

Thanks

**Reply by webbooo** [reply]
hey @snibgo I was able to use something like the below, not perfect but it works lol - thanks again for your help

`for %%F in (c:\destination\*.png) do magick convert %%F ( C:\source\%%~nF.png -virtual-pixel None +distort SRT 0,0,1,22,0,0 -repage "+55+25^!" ) -compose Over -layers merge C:\outcome\%%~nF.png`

**Comment by snibgo** [comment]
See also my latest reply to https://github.com/ImageMagick/ImageMagick/discussions/5114
